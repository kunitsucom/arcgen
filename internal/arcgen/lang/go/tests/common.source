package main

type (
	// User is a user.
	//
	// dbtest: table: `Users`
	User struct {
		// ID is a user ID.
		ID string     `dbtest:"Id"`
		// Name is a user name.
		Name string   `dbtest:"Name"`
		// Email is a user email.
		Email string  `dbtest:"Email"`
		// Age is a user age.
		Age int       `dbtest:"Age"`
		// Ignore is a ignore field.
		Ignore string `dbtest:"-"`
	}

	// Users is a slice of User.
	//
	Users []*User

	// dbtest: table: `InvalidUsers`
	InvalidUser struct {
		ID string
	}
)
