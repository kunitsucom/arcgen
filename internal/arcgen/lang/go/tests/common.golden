// Code generated by arcgen. DO NOT EDIT.
//
// source: tests/common.source

package main

func (s User) GetTableName() string {
	return "Users"
}

func (s User) GetColumnNames() []string {
	return []string{"Id", "Name", "Email", "Age"}
}

func (s User) GetColumnName_ID() string {
	return "Id"
}

func (s User) GetColumnName_Name() string {
	return "Name"
}

func (s User) GetColumnName_Email() string {
	return "Email"
}

func (s User) GetColumnName_Age() string {
	return "Age"
}

func (s Group) GetColumnNames() []string {
	return []string{"Id", "Name"}
}

func (s Group) GetColumnName_ID() string {
	return "Id"
}

func (s Group) GetColumnName_Name() string {
	return "Name"
}
